### PR TITLE
Introduce monitoring with prometheus/ grafana

### PR DESCRIPTION
**Pull request by bokuanT**
@thamruicong 

**Description**
Introduce prometheus with cadvisor and node exporter to collect metrics on containers and the VM itself. Also modified backend server to collect api req-res metrics to be scraped by prometheus. Metrics are visualised with a few dashboards in grafana. Currently I plan to make the data accessible by doing ssh port forwarding to the vm so the monitoring stuff isnt't exposed.

Docker compose of staging is already updated to this, which introduces a network in the docker compose so prometheus can communicate with the backend. TODO: update the docker compose of production

This PR's deployment will just add the /metrics endpoint to the backend. I still need to manually run the script to deploy the monitoring on the VM - to look into adding a GH action CD to auto-run this script

**Other Information**

**Checklist**

- [X] My pull request has a descriptive title (not a vague title like `Update README.md`)

- [X] My pull request targets the `master` branch of the repository only

- [X] My commits follows these [commit message guidelines](https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53)

- [ ] I added tests for the changes I made (if applicable)

- [ ] I added or updated documentation (if applicable)

- [X] I have ran the project and its tests locally and verified that there are no visible errors

- [ ] Includes DB Migration
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested

### CODE REVIEW

File: packages/frontend/local_dev_docker/Dockerfile.dev
Comment (bokuanT) [inline packages/frontend/local_dev_docker/Dockerfile.dev:1]: If you want to build and run the app locally, the current dockerfile doesn't work as it doesn't forward your api calls to the backend. This configures the nginx server to do the forwarding if you run the containerised built app locally

**Comment by bokuanT** [comment]
will try to add GH action CD

File: .github/workflows/deploy-monitoring-staging.yaml
Comment (bokuanT) [inline .github/workflows/deploy-monitoring-staging.yaml:1]: added new webhook on staging

File: README.md [hunk 1]
```diff
@@ -32,3 +32,22 @@ At project root:
...
+1. Ensure database has been seeded and set up
+
+2. Create a `.env.docker`:
+
+```
+POSTGRES_USER=admin
+POSTGRES_PASSWORD=admin
+POSTGRES_DB=trofos
+DATABASE_URL="postgresql://admin:admin@postgres:5432/trofos?schema=public"
+trofos?schema=public"
```
Comment (thamruicong) [inline README.md:47]: extra line here?

**Review by thamruicong — COMMENTED** [review]
lgtm

**Pull request merged**